libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -105,14 +110,11 @@ string YulUtilFunctions::copyToMemoryFunction(bool _fromCalldata)
 					{
 						mstore(add(dst, i), mload(add(src, i)))
 					}
-					if gt(i, length)
-					{
-						// clear end
-						mstore(add(dst, length), 0)
-					}
+					<?cleanup>mstore(add(dst, length), 0)</cleanup>
```

Comment:
This line always made me uncomfortable: it ends up writing beyond what it's supposed to. It should only be managing memory from `[dst, dst + length)` at least from the signature of the function, but it ends up creeping into `[dst + length, dst + length + 32)` now.

Reply:
The plan for after the release is to see if we can't just inline the function into all of its use cases (it's a weird helper function anyways and it only has four use cases, in which it should be apparent that the overwrite is fine).

Reply:
And if it stays, it and everything that uses it transitively should have scary comments and docstring pointing out the potential risk.